SwiftDungeon/AttackButton.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  AttackButton.swift
+//  SwiftDungeon
+//
+//  Created by Puntillo Andrew J. on 3/19/18.
+//  Copyright © 2018 Toro Juan D. All rights reserved.
+//
+
+import Foundation
+import SpriteKit
+
+class AttackButton {
```

Comment:
Should extend sprite node so you can override methods in it.
```
class AttackButton: SKSpriteNode {

Reply:
Many of these core classes are designed to be extended and overridden to get full functionality.